[#654] Add AbstractEditorDoubleClickTextSelectionTest abstract class.

miklossy opened the pull request:
- Extend the 'Xtext UI Testing' test infrastructure by the
AbstractEditorDoubleClickTextSelectionTest abstract class to provide a
base infrastructure for testing a customized DoubleClickStrategyProvider
behaviour.

Signed-off-by: Tamas Miklossy <miklossy@itemis.de>

szarnekow commented on org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractEditorDoubleClickTextSelectionTest.xtend:
```diff
@@ -0,0 +1,99 @@
+				must contain the �c� symbols indicating the current cursor position!
+			''')
+		}
+		cursorPosition
```
It's a matter of taste, but explicit returns read better in code blocks.

miklossy replied:
I personally prefer implicit return and use it wherever as possible. What do @cdietrich and @kthoms think?

cdietrich replied:
if i use implict returns in combination with a non single line statement i specifiy a explicit return type

holgerschill replied:
I prefer explicit returns always since it reads better.

cdietrich commented on org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractEditorDoubleClickTextSelectionTest.xtend:
```diff
@@ -0,0 +1,99 @@
+	 *            in the Xtext editor on the current cursor position.
+	 */
+	def assertSelectedTextAfterDoubleClicking(CharSequence it, String expected) {
+		
```
formatting looks wired

miklossy commented:
I updated the commit in the PR and implemented the following improvements:

- Use injection of the XtextEditorInfo object to access the editor id.
- Specify an explicit return type in method signatures using implict returns in combination with a non single line statement.
- Change the default special symbol from '<|>' to '|' to improve the readability of the tests.
- Improve the javadoc formatting.

Thank you both for your feedback!
